### ISSUE TITLE
Notification title shows user email instead of user name

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#19716]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Write a message to someone in NewDot
2. Unfocus the native NewDot window and wait for them to reply
3. Note that the pop-up notification show's the user's email address instead of their name

## Expected Result: 
I would expect to see the user's name if they've added it
## Actual Result: 
I see the user's email address only

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.19-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

![image (1)](https://github.com/Expensify/App/assets/43996225/761bbc2b-8f67-4c5b-9609-7981134d5f1b)

https://github.com/Expensify/App/assets/43996225/d6c5e64a-55a2-45fe-929b-521e919ddcd4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @greg-schroeder 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684968338273599

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by shivansh9007** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
User email displayed in notification even if users display name is available.
### What is the root cause of that problem?
While pushing the notification we are picking up the email always.
### What changes do you think we should make in order to solve the problem?

Pickup user's Display Name using ReportUtils.getDisplayNameForParticipant()  and assign it to title [here](https://github.com/Expensify/App/blob/main/src/libs/Notification/LocalNotification/BrowserNotifications.js#L118) 
Now line 118 looks like
       **title = ReportUtils.getDisplayNameForParticipant(plainTextPerson,false);**

### What alternative solutions did you explore? (Optional)
None

**Comment by tienifr** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Email is displayed in notification instead of displayName

### What is the root cause of that problem?

In https://github.com/Expensify/App/blob/06192acc94f8fa31f95adb76f57bd9f4145bdf11/src/libs/Notification/LocalNotification/BrowserNotifications.js#L107-L108

We're getting the email from reportActions

### What changes do you think we should make in order to solve the problem?

We should apply ReportUtils.getDisplayNameForParticipant() for each person 

```
        const plainTextPerson = _.map(person, (f) => ReportUtils.getDisplayNameForParticipant(f.text)).join();
```

We also should update the push notification for native platform to show the displayName instead of email as well

### What alternative solutions did you explore? (Optional)

None

**Comment by shivansh9007** [comment]
Updated Proposal: After testing @tienifr 's proposal I found another issue in the code. Updating the proposal to address the notification issue with Chat Rooms (group chats)
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

User email displayed in notification even if users display name is available.

### What is the root cause of that problem?

There are two issues here

1. While pushing the notification we are picking up the email always. (As mentioned originally [here](https://github.com/Expensify/App/issues/19716#issuecomment-1565248990)
2. The chatType from API response is always an empty String ""; Hence the **notification card for ChatRooms is never display as intended** and it falls back to the notification card designed for user chats.


### What changes do you think we should make in order to solve the problem?
1. Update [here](https://github.com/Expensify/App/blob/main/src/libs/Notification/LocalNotification/BrowserNotifications.js#L108). To get messages senders name if available with.
        const plainTextPerson = _.map(person, (f) => ReportUtils.getDisplayNameForParticipant(f.text)).join();

2. Ensure the Backend is updated to send proper chatType Value. 

### What alternative solutions did you explore? (Optional)
None.

###References & Screenshots:
<img width="1007" alt="Screenshot 2023-05-29 at 1 47 18 PM" src="https://github.com/Expensify/App/assets/22397210/5be5e194-9313-4d21-ba5a-270bea06253a">

Mocked chatType and tested :
<img width="370" alt="Screenshot 2023-05-29 at 2 03 29 PM" src="https://github.com/Expensify/App/assets/22397210/f71b9d20-214c-4e54-9eba-a796310ba4a6">

**Comment by Julesssss** [comment]
This is also occurring on native Android, so likely other platforms too (iOS / Desktop).

![image](https://github.com/Expensify/App/assets/10736861/21ffb0f4-37d8-467f-8787-63b48927f771)

**Comment by Julesssss** [comment]
I think this is likely an internal issue.

**Comment by arosiclair** [comment]
PR is posted

**Comment by arosiclair** [comment]
Waiting on deploy

**Comment by arosiclair** [comment]
Deployed to prod! Since this was reported, fixed and reviewed internally, no payment is needed!
